### PR TITLE
Composable refactor

### PR DESCRIPTION
**Pull request by GavinHuttley**
Fixes to `available_apps()`

### CODE REVIEW

File: src/cogent3/app/__init__.py [hunk 1]
```diff
@@ -33,24 +33,23 @@ def _get_app_attr(name, is_composable):
...
 
-    mod = importlib.import_module(f"{__name__}.composable")
-    rows.append(
-        _get_app_attr(
-            f"{mod.__name__}.user_function", issubclass(user_function, Composable)
-        )
-    )
-    header = ["module", "name", "composable", "doc", "outputs", "data type"]
-    return Table(header, rows)
+    rows = [_get_app_attr(app, __app_registry.get(app)) for app in __app_registry]
```
Comment (Nick-Foto) [inline src/cogent3/app/__init__.py:53]: In `__app_registry`, keys are `get_object_provenance(klass)`. But here we use `app` as key. Are these equal?

File: src/cogent3/app/__init__.py [hunk 2]
```diff
@@ -33,24 +33,23 @@ def _get_app_attr(name, is_composable):
         name,
         is_composable,
         obj.__doc__,
-        _types["_data_types"],
-        _types["_return_types"],
+        ", ".join(sorted(_types["_data_types"])),
+        ", ".join(sorted(_types["_return_types"])),
```
Comment (Nick-Foto) [inline src/cogent3/app/__init__.py:37]: `sorted` changes order, but we may need order later.

Comment (Nick-Foto) [reply src/cogent3/app/__init__.py:37]: We may need order to find type of the first argument of `main` function.

File: src/cogent3/app/__init__.py (hunk 1 above)
Comment (GavinHuttley) [reply src/cogent3/app/__init__.py:53]: I'm iterating over the keys in the dict, so yes.

File: src/cogent3/app/__init__.py (hunk 2 above)
Comment (GavinHuttley) [reply src/cogent3/app/__init__.py:37]: Not in the table. Note, the original app attributes are `frozenset()` (thus immutable).

These lists are created only to produce this table.

File: src/cogent3/app/__init__.py (hunk 1 above)
Comment (GavinHuttley) [reply src/cogent3/app/__init__.py:53]: Actually, there's no need to do the `__app_registry.get(app)`. The value must exist! Fixing that now.

File: src/cogent3/app/__init__.py (hunk 2 above)
Comment (Nick-Foto) [reply src/cogent3/app/__init__.py:37]: OK, Now I understand it is only for this table.

File: src/cogent3/app/__init__.py (hunk 1 above)
Comment (Nick-Foto) [reply src/cogent3/app/__init__.py:53]: We have used `get` function for dict a lot of times, so we forget to use normal usage of key indexing.

**Pull request merged**